### PR TITLE
fix(typeahead): add more states while searching

### PR DESCRIPTION
**Pull request by frassinier**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the new behavior?**
add special behavior for searching and no result cases

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: src/Typeahead/Typeahead.component.js [hunk 1]
```diff
@@ -62,6 +69,7 @@ function Typeahead({ onToggle, icon, position, ...rest }) {
 			itemsList: theme['items-list'],
 			sectionContainer: theme['section-container'],
 		},
+		items: rest.items || [],
```
Comment (jsomsanith) [inline src/Typeahead/Typeahead.component.js]: Why `|| []` ? The default props is `null`, and [] is the base info to render the `no result`message

File: src/Typeahead/subRenderers.js [hunk 2]
```diff
@@ -14,11 +15,9 @@ export const renderInputComponent = (props) => {
 		...rest,
 	} = props;
 
-	const renderedIcon = icon && (
-		<div className={theme['icon-cls']}>
-			<Icon name={icon.name} title={icon.title}/>
-		</div>
-	);
+	const renderedIcon = icon && (<div className={theme['icon-cls']}>
+			<Icon name={icon.name} title={icon.title} />
```
Comment (jsomsanith) [inline src/Typeahead/subRenderers.js]: Linter error

File: src/Typeahead/subRenderers.js [hunk 3]
```diff
@@ -48,6 +49,27 @@ renderInputComponent.propTypes = {
 	}),
 };
 
+export const renderItemsContainer = (items, noResultText, searching, searchingText) => (props) => {
+	const { className, ...restProps } = props; // eslint-disable-line react/prop-types
```
Comment (jsomsanith) [inline src/Typeahead/subRenderers.js]: Remove this linter disable and create a react functional component (in the same file) with defined PropTypes. The renderItemsContainer will just instanciate it
export const renderItemsContainer = (items, noResultText, searching, searchingText) => (props) => <ItemContainer {...props} items={items} noResultText=... />

File: stories/AppHeaderBar.js [hunk 4]
```diff
@@ -196,6 +196,22 @@ decoratedStories
...
+			search: {
+				icon: {
+					name: 'fa fa-search',
+				},
+				position: 'right',
+				value: 'le',
+				searching: true,
+				onBlur: action('onBlur'),
+				onChange: action('onChange'),
+				onSelect: action('onSelect'),
```
Comment (jsomsanith) [inline stories/AppHeaderBar.js]: Remove that unneeded prop

File: stories/AppHeaderBar.js [hunk 5]
```diff
@@ -211,4 +227,20 @@ decoratedStories
...
+			search: {
+				icon: {
+					name: 'fa fa-search',
+				},
+				position: 'right',
+				value: 'le',
+				items: [],
+				onBlur: action('onBlur'),
+				onChange: action('onChange'),
+				onSelect: action('onSelect'),
```
Comment (jsomsanith) [inline stories/AppHeaderBar.js]: Remove that unneeded prop

File: src/Typeahead/Typeahead.scss [hunk 6]
```diff
@@ -50,71 +47,84 @@ $tc-typeahead-items-list-shadow-color: #CCC;
...
-		border-radius: 4px;
-		box-shadow: 1px 1px 5px 1px $tc-typeahead-items-list-shadow-color;
-		max-height: 70vh;
-		overflow-y: auto;
+.is-searching,
+.no-result,
+.container-open .items-container {
+	display: block;
+	padding-top: 1px;
+	padding-bottom: 1px;
```
Comment (jsomsanith) [inline src/Typeahead/Typeahead.scss]: Those paddings makes the box quite small, and not aligned with guideline

File: src/Typeahead/Typeahead.scss [hunk 7]
```diff
@@ -50,71 +47,84 @@ $tc-typeahead-items-list-shadow-color: #CCC;
...
+}
+
+.is-searching {
+	display: flex;
+	overflow: hidden;
```
Comment (jsomsanith) [inline src/Typeahead/Typeahead.scss]: I don't see why this is needed ?

File: src/Typeahead/Typeahead.scss [hunk 8]
```diff
@@ -50,71 +47,84 @@ $tc-typeahead-items-list-shadow-color: #CCC;
...
 
 .item {
-	cursor: pointer;
 	margin-bottom: 0;
-	padding: 14px;
-
-	.item-title {
-		font-weight: normal;
-	}
+	padding: 15px;
```
Comment (jsomsanith) [inline src/Typeahead/Typeahead.scss]: Would it be interesting to express that with our theme variables ?

File: src/Typeahead/Typeahead.scss [hunk 9]
```diff
@@ -50,71 +47,84 @@ $tc-typeahead-items-list-shadow-color: #CCC;
...
 
-/******************************************** section **************************************/
 .section-header {
 	align-items: center;
 	color: $tc-typeahead-section-header-color;
 	border-top: 1px solid $tc-typeahead-section-border-color;
 	border-bottom: 1px solid $tc-typeahead-section-border-color;
 	margin-top: -1px;
-	padding: 14px;
+	padding: 15px;
```
Comment (jsomsanith) [inline src/Typeahead/Typeahead.scss]: Would it be interesting to express that with our theme variables ?

File: src/Typeahead/Typeahead.scss [hunk 10]
```diff
@@ -50,71 +47,84 @@ $tc-typeahead-items-list-shadow-color: #CCC;
...
-		font-weight: normal;
-	}
+	padding: 15px;
+	cursor: pointer;
 
-	.item-description {
+	&-description {
 		color: $tc-typeahead-item-description-color;
-		font-size: 1.3rem;
+		font-size: 13px;
```
Comment (jsomsanith) [inline src/Typeahead/Typeahead.scss]: Why in px ?

File: src/Typeahead/Typeahead.scss [hunk 11]
```diff
@@ -50,71 +47,84 @@ $tc-typeahead-items-list-shadow-color: #CCC;
...
-	padding: 14px;
+	padding: 15px;
 	display: flex;
 
-	.section-header-title {
-		margin-left: 7px;
-		font-size: 1.2rem;
+	&-title {
+		padding: 0 10px;
+		font-size: 12px;
```
Comment (jsomsanith) [inline src/Typeahead/Typeahead.scss]: Why in px ?

File: src/Typeahead/Typeahead.scss [hunk 12]
```diff
@@ -50,71 +47,84 @@ $tc-typeahead-items-list-shadow-color: #CCC;
...
 	margin-top: -1px;
-	padding: 14px;
+	padding: 15px;
 	display: flex;
 
-	.section-header-title {
-		margin-left: 7px;
-		font-size: 1.2rem;
+	&-title {
+		padding: 0 10px;
```
Comment (jsomsanith) [inline src/Typeahead/Typeahead.scss]: Would it be interesting to express that with our theme variables ?

File: src/Typeahead/Typeahead.component.js (hunk 1 above)
Comment (frassinier) [inline src/Typeahead/Typeahead.component.js]: Cause items cannot be null within `react-autowathever` and I need to distinguish `no result` from `not looking for results` states

File: src/Typeahead/Typeahead.scss (hunk 6 above)
Comment (frassinier) [inline src/Typeahead/Typeahead.scss]: These values come from the guidelines in fact.

File: src/Typeahead/Typeahead.scss (hunk 7 above)
Comment (frassinier) [inline src/Typeahead/Typeahead.scss]: Because of circular component which rotates and implies overflow.

**Pull request merged**